packages/block-library/src/paragraph/edit.js

```diff
@@ -244,8 +244,8 @@ class ParagraphBlock extends Component {
 					} }
 					unstableOnSplit={ this.splitBlock }
 					onMerge={ mergeBlocks }
-					onReplace={ this.onReplace }
-					onRemove={ () => onReplace( [] ) }
+					onReplace={ this.props.onReplace && this.onReplace }
+					onRemove={ this.props.onReplace && ( () => onReplace( [] ) ) }
```

Comment:
We have the prop destructured already (it's referenced on this same line).

```suggestion
					onRemove={ onReplace && ( () => onReplace( [] ) ) }
```